### PR TITLE
environment.py: Detect all mips* architectures

### PR DESCRIPTION
**Pull request by persianpros**
We have more than those values, like:

mipsel
mipsel-nf
mips32el
mips33el-nf

So lets just detect them all.

### CODE REVIEW

File: mesonbuild/environment.py [hunk 1]
```diff
@@ -377,8 +377,8 @@ def detect_cpu_family(compilers: CompilersDict) -> str:
         trial = 'x86_64'
     elif trial in {'sun4u', 'sun4v'}:
         trial = 'sparc64'
-    elif trial in {'mipsel', 'mips64el'}:
-        trial = trial.rstrip('el')
+    elif trial.startswith('mips'):
+        trial = 'mips'
```
Comment (eli-schwartz) [inline mesonbuild/environment.py:381]: This no longer returns 'mips64' in situations where it used to, so I'm not sure how you expect this to work.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: mesonbuild/environment.py (hunk 1 above)
Comment (persianpros) [reply mesonbuild/environment.py:381]: Oh I forgot that I just tested the 32 bit, let me send another one soon.

**Comment by eli-schwartz** [comment]
Please don't close PRs like that whenever changes are requested. Just push an update to the current PR.

**Comment by persianpros** [comment]
Sure I do, I just didn't keep my fork for the last one so I had to fork again.

Sorry for any inconvenience.

**Pull request closed without merging**